docs/infrastructure.md

```diff
@@ -0,0 +1,206 @@
+# Infrastructure
```

Comment:
Could sync these docs with the wiki.  Easier to read that way IMO:

https://github.com/level12/coppy/blob/main/.github/workflows/docs.yaml